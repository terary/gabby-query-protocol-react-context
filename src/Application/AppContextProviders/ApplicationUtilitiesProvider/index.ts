import ApplicationUtilitiesProvider, {
  ApplicationUtilitiesContext,
  useTestUtilities,
} from "./ApplicationUtilitiesProvider";
import type { ApplicationUtilitiesType } from "./type";

export default ApplicationUtilitiesProvider;
export { ApplicationUtilitiesContext, useTestUtilities };
export type { ApplicationUtilitiesType };
