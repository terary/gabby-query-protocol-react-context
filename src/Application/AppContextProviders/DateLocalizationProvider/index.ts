import DateLocalizationProvider from "./Provider";

export default DateLocalizationProvider;
