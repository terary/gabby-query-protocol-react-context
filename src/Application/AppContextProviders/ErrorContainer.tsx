import { ReactNode } from "react";
import { ErrorBoundary } from "react-error-boundary";

function ErrorFallback({ error }: { error: Error }) {
  return (
    <div role="alert">
      <p>Something went wrong:</p>
      <pre>{error.message}</pre>
    </div>
  );
}

export const ErrorContainer = ({ children }: { children: ReactNode }) => {
  return <ErrorBoundary FallbackComponent={ErrorFallback}>{children}</ErrorBoundary>;
};
