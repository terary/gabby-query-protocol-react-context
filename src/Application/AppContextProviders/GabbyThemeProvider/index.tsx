import { GabbyThemeProvider } from "./Provider";
export default GabbyThemeProvider;
