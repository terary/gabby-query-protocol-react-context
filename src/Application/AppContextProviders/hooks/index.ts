import { useApplicationUtilities } from "./useApplicationUtilities";

export { useApplicationUtilities };
