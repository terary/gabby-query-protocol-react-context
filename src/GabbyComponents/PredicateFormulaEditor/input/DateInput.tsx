import { ExperimentalDateInputWithAppContext } from "../../common/experimental/ExperimentalDateInput";

const noopIsValid = (newValue: unknown) => true;

interface Props {
  formatMask: string;
  id: string;
  isValid?: (value: unknown) => boolean;
  label: string;
  name: string;
  onChange: (value: string) => void; // we'll allow empty strings
  value: string;
}
export const DateInput = ({
  formatMask,
  id,
  isValid = noopIsValid,
  label,
  name,
  onChange,
  value,
}: Props): JSX.Element => {
  return (
    <ExperimentalDateInputWithAppContext
      //      formatMask={getCurrentLocale().dateFormatMap}
      id={id}
      isValid={isValid}
      label={label}
      name={name}
      onChange={onChange}
      value={value}
    />
  );
};
