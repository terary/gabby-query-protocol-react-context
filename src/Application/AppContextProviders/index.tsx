import React from "react";
import { TSupportedLocales } from "..";
import ApplicationUtilitiesProvider, {
  useTestUtilities,
} from "./ApplicationUtilitiesProvider";
import { ErrorContainer } from "./ErrorContainer";
import DateLocalizationProvider from "./DateLocalizationProvider";
import GabbyThemeProvider from "./GabbyThemeProvider";

interface IAppProviderParameters {
  children?: React.ReactNode;
  initialLocaleCode?: TSupportedLocales;
}

export const AppContextProviders = ({
  initialLocaleCode,
  children,
}: IAppProviderParameters) => {
  return (
    <ErrorContainer>
      {" "}
      {/* Error container may have issue with hooks, requires testing */}
      <ApplicationUtilitiesProvider initialLocaleCode={initialLocaleCode}>
        <GabbyThemeProvider>
          <DateLocalizationProvider>{children}</DateLocalizationProvider>
        </GabbyThemeProvider>
      </ApplicationUtilitiesProvider>
    </ErrorContainer>
  );
};

export { useTestUtilities };
