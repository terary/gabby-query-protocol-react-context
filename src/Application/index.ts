import { SUPPORTED_LOCALE_CODES_ARRAY, SUPPORTED_LOCALE_DICTIONARY } from "./i18n";
import { PageContentWrapper } from "./PageContentWrapper";
import { useApplicationUtilities } from "./AppContextProviders/hooks";
import { TSupportedLocales } from "./i18n";
import { customStyles } from "./custom-styles";
export {
  customStyles,
  PageContentWrapper,
  useApplicationUtilities,
  SUPPORTED_LOCALE_CODES_ARRAY,
  SUPPORTED_LOCALE_DICTIONARY,
};

export type { TSupportedLocales };
