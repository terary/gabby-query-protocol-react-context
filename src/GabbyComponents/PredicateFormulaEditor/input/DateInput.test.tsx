import React from "react";
import renderer from "react-test-renderer";
import { DateInput } from "./DateInput";
import { BrowserRouter } from "react-router-dom";
import { AppContextProviders } from "../../../Application/AppContextProviders";

const props = {
  formatMask: "__/__/____",
  id: "dateInput-id",
  isValid: (value: unknown) => true,
  label: "dateInput-label",
  name: "dateInput-name",
  onChange: (value: string) => {}, // we'll allow empty strings
  value: "1969-07-20",
};

it("renders correctly", () => {
  const tree = renderer
    .create(
      <AppContextProviders>
        <BrowserRouter>
          <DateInput {...props} />
        </BrowserRouter>
      </AppContextProviders>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
