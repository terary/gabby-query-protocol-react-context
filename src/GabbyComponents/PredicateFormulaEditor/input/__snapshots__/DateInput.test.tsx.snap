// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  " ",
  <div
    className="Toastify"
  />,
  " ",
  <div
    className="Toastify"
  />,
  <div
    className="MuiFormControl-root MuiTextField-root css-136h9lv-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
    >
      dateInput-label
    </label>
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        aria-label="Choose date, selected date is 19 jul 1969"
        aria-readonly={true}
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        id="dateInput-id"
        name="dateInput-name"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        readOnly={true}
        required={false}
        type="text"
        value="19/07/1969"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-dhh0px"
        >
          <span>
            dateInput-label
          </span>
        </legend>
      </fieldset>
    </div>
  </div>,
]
`;
